### ISSUE TITLE
exclude_paths is not working as expected

### ISSUE THREAD

**Issue by kkartch** [codeclimate/codeclimate-fixme#28]
I'm experiencing the same issue as https://github.com/codeclimate/codeclimate-fixme/issues/19 which is closed.  I'm opening a new one in case you do not want to reopen the previous bug.

I'm evaluating this for our private repos on a trial, but not getting good results due to this.  My .codeclimate.yml:

``` yaml
engines:
  duplication:
    enabled: true
    config:
      languages:
      - go
  fixme:
    enabled: true
  gofmt:
    enabled: true
  golint:
    enabled: true
  govet:
    enabled: true
ratings:
  paths:
  - "**.go"
exclude_paths:
- Godeps/**/*
```

Trimmed output is:

```
Starting analysis

== Godeps/_workspace/src/github.com/ajg/form/decode.go (2 issues) ==
18: exported func NewDecoder returns unexported type *form.decoder, which can be annoying to use [golint]
282: TODO found [fixme]

== Godeps/_workspace/src/github.com/ajg/form/encode.go (1 issue) ==
20: exported func NewEncoder returns unexported type *form.encoder, which can be annoying to use [golint]

== Godeps/_workspace/src/github.com/ajg/form/node.go (2 issues) ==
34: TODO found [fixme]
39: TODO found [fixme]

== Godeps/_workspace/src/github.com/ajg/form/pre-commit.sh (2 issues) ==
3: TODO found [fixme]
4: FIXME found [fixme]
```

I have also tried the many different variants of the exclude_paths selection with Godeps/, Godeps/\*\*, Godeps/\*\*/\*.  With quotes/without quotes.

Latest version of codeclimate and latest docker images:

``` bash
codeclimate --version
0.20.0
```

``` bash
docker images | grep codeclimate
codeclimate/codeclimate-csslint       latest              468d26f5e706        24 hours ago        133.5 MB
codeclimate/codeclimate-eslint        latest              bba6f4524ffe        4 days ago          90.44 MB
codeclimate/codeclimate               latest              62eca8db5a53        4 days ago          95.09 MB
codeclimate/codeclimate-fixme         latest              868840c0a743        5 days ago          211 MB
codeclimate/codeclimate-duplication   latest              04e9cc6b0cc7        8 days ago          811 MB
codeclimate/codeclimate-govet         latest              88185bc0e1b0        5 weeks ago         217.6 MB
codeclimate/codeclimate-golint        latest              964303486d6d        5 weeks ago         11.22 MB
codeclimate/codeclimate-gofmt         latest              acae6da3479b        5 weeks ago         218.3 MB
```

**Comment by ABaldwinHunter** [comment]
@kkartch Thanks for the feedback!

This issue is no longer expected to be a problem for the `fixme` engine.

`Godeps/` or `Godeps/**/*` both are fine for your exclude path format.

Could you try running `docker pull codeclimate/codeclimate-fixme` to make sure you have the most up to date engine, with correct exclude paths, and see if that fixes the Fixme results?

That said, the [golint engine](https://github.com/codeclimate-community/codeclimate-golint/blob/master/codeclimate-golint.go#L30) needs to be updated to support receiving `include_paths` (internal derivative of `exclude_paths` from yaml).

Could you open this issue there? Thanks!

**Comment by wfleming** [comment]
Unrelated to the engine bugs here, but the duplication engine does not support go, so that engine isn't actually going to analyze anything right now. It should probably raise an error for a case like this, but doesn't right now.

**Comment by wfleming** [comment]
@kkartch I've released new versions of our go engines to support the newer way our platform processes exclude paths: unfortunately they had fallen a bit behind. If you run `codeclimate engines:install` from the project directory, it will install those updates. That should fix the erroneous `golint` issues being reported, so please let us know if it doesn't.

You do appear to be on the latest version of the fixme engine, but we're having a hard time reproducing that bug. Can you please confirm that your `.codeclimate.yml` still has the same contents you posted above & you're still seeing these? If you are, this might be caused by some unusual collection of file/path names specific to your project, so any more details about your project you can provide would be appreciated. You can email me at will@codeclimate.com if there are details you'd feel comfortable sharing with us but would rather not post publicly.

**Comment by kkartch** [comment]
I updated my engines this morning, and all looks great.  I believe I had the latest fixme engine yesterday, but for some reason it is working this morning.

Unfortunately, the results are not all that incredibly helpful for Go.  I got a lot of value out of CodeClimate for Ruby projects, but it doesn't look like it's quite there for Go yet.  I'll keep checking in on it.

Thank for you both for your help.
